engine/schema/src/main/java/com/cloud/vm/dao/VMInstanceDaoImpl.java

```diff
@@ -1041,6 +1042,8 @@ public Pair<List<VMInstanceVO>, Integer> listByVmsNotInClusterUsingPool(long clu
         sc.setParameters("vmStates", State.Starting, State.Running, State.Stopping, State.Migrating, State.Restoring);
         sc.setJoinParameters("volumeSearch", "poolId", poolId);
         sc.setJoinParameters("hostSearch2", "clusterId", clusterId);
-        return searchAndCount(sc, null, false);
+        List<VMInstanceVO> vms = search(sc, null);
+        List<VMInstanceVO> uniqueVms = vms.stream().distinct().collect(Collectors.toList());
+        return new Pair<>(uniqueVms, uniqueVms.size());
```

Comment:
make sense :+1: 

@abh1sar 
code looks good. have you tested it ? (vm has multiple disks on a pool)

Reply:
Yes, it was returning multiple entries for the same VM like this

![Screenshot from 2024-05-22 09-51-24](https://github.com/apache/cloudstack/assets/63767682/c672d51b-6a55-4c38-b3f5-bce3c338f32e)

Reply:
good finding !